Classes/Plugin/Search.php

```diff
@@ -117,19 +117,26 @@ protected function addCurrentDocument()
             && \TYPO3\CMS\Core\Utility\MathUtility::canBeInterpretedAsInteger($this->piVars['id'])
         ) {
             $this->loadDocument();
-            // Get document's UID or parent ID.
+            // Get document's UID
             if ($this->doc->ready) {
-                return '<input type="hidden" name="' . $this->prefixId . '[id]" value="' . ($this->doc->parentId > 0 ? $this->doc->parentId : $this->doc->uid) . '" />';
```

Comment:
Why should be set the parentId here?

Reply:
In the past, users always wanted to search within an intellectual work, not just within an arbitrary part of it. At least that is true for multivolume works and multi-part manuscripts. That's why we always checked for a `parentId` and searched within all associated children (i. e. volumes).

I think, the correct way would be to get the actual document type and check if it is configured as "toplevel". If it is, it's ok to search just within that document and all of its descendants. If it is not, we need to get the ancestor (not just parent) "toplevel" document and search within that and all of its descendants (not just children).

Basically that's what the current implementation did, but without explicitly checking the document type and without respecting more than two hierarchical levels. Since newspapers have more than two hierarchical levels, it's not sufficient anymore to just check for the `parentId`. Instead we have to go up in the hierarchy until we find a "toplevel" ancestor and then perform the search with this "toplevel" ancestor's UID and all descendants.